kge/model/kg2e.py

```diff
@@ -0,0 +1,105 @@
+import torch
+from kge import Config, Dataset
+from kge.model.kge_model import KgeModel, RelationalScorer
+from kge.job import Job
+from torch.nn import functional as F
+
+
+def score_kl(mean, ent_cov, rel_cov):
+    return -(
+        torch.sum((1 / rel_cov) * ent_cov, dim=-1)
+        + torch.sum(mean * mean * (1 / rel_cov), dim=-1)
+        - torch.sum(torch.log(ent_cov), dim=-1)
+        + torch.sum(torch.log(rel_cov), dim=-1)
+    )
+
+
+def score_el(mean, ent_cov, rel_cov):
+    return -(
+        torch.sum(mean * mean * (1 / (ent_cov + rel_cov)), dim=-1)
+        + torch.sum(torch.log(ent_cov + rel_cov), dim=-1)
+    )
+
+
+class KG2EScorer(RelationalScorer):
+    def __init__(self, config: Config, dataset: Dataset, configuration_key=None):
+        super().__init__(config, dataset, configuration_key)
+        score_metric = self.get_option('score_function')
+        if score_metric.upper() == 'KL':
+            self.score_func = score_kl
+        elif score_metric.upper() == 'EL':
+            self.score_func = score_el
+        else:
+            raise ValueError('No score function implemented for metric {}'.format(score_metric))
+
+    def _score(self, s_emb, p_emb, o_emb):
+        s_mean, s_cov = torch.chunk(s_emb, 2, dim=1)
+        p_mean, p_cov = torch.chunk(p_emb, 2, dim=1)
+        o_mean, o_cov = torch.chunk(o_emb, 2, dim=1)
+        # compute the overall mean and entity covariance matrix
+        mean = s_mean - p_mean - o_mean
+        ent_cov = s_cov + o_cov
+
+        return self.score_func(mean, ent_cov, p_cov)
+
+    def score_emb(self, s_emb, p_emb, o_emb, combine: str):
+
+        n = p_emb.size(0)
+        if combine == "spo":
+            # n = n_s = n_p = n_o
+            out = self._score(s_emb, p_emb, o_emb)
+        else:
+            return super().score_emb(s_emb, p_emb, o_emb, combine)
+
+        return out.view(n, -1)
+
+
+class KG2E(KgeModel):
+    def __init__(
+        self,
+        config: Config,
+        dataset: Dataset,
+        configuration_key=None,
+        init_for_load_only=False,
+    ):
+        self._init_configuration(config, configuration_key)
+        self._dim = self.get_option("entity_embedder.dim")
+        self.set_option("entity_embedder.dim", self._dim * 2, log=True)
+        if self.get_option("relation_embedder.dim") < 0:
+            self.set_option("relation_embedder.dim",  self._dim * 2, log=True)
```

Comment:
We can load the model by using the below command: 
```
import torch
from kge.model import KgeModel
from kge.util.io import load_checkpoint

# download link for this checkpoint given under results above
checkpoint = load_checkpoint('fb15k-237-rescal.pt')
model = KgeModel.create_from(checkpoint)
```
But because of the changes in dimensions, we can't load the model. 
So I think it is better to change like the below:
```
        if not init_for_load_only:
            self.set_option("entity_embedder.dim", self.dim * 2, log=True)
            if self.get_option("relation_embedder.dim") < 0:
                self.set_option("relation_embedder.dim", self.dim * 2, log=True)
```